.github/workflows/nightly_release.yml

```diff
@@ -0,0 +1,72 @@
+name: SalesforceCommerceCloud/pwa-kit/nightly_release
+on:
+  workflow_dispatch:
+  schedule:
+    # Run every day at 12am (PST) - cron uses UTC times
+    - cron: "0 8 * * *"
+
+jobs:
+  create_nightly_release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get Timestamp
+        run: |-
+          echo "release_timestamp=$(date +'%m%d%Y%H%M%S')" >> "$GITHUB_ENV"
+
+      - name: Get monorepo version
+        run: |-
+          version=`cat package.json | jq -r ".version" | cut -d "-" -f 1`
+          echo "monorepo_version_base=$version" >> "$GITHUB_ENV"
+          version_major=`echo "$version" | cut -d "." -f 1`
+          echo "monorepo_version_major=$version_major" >> "$GITHUB_ENV"
+
+      - name: Get retail-react-app version
+        run: |-
+          version=`cat packages/template-retail-react-app/package.json | jq -r ".version" | cut -d "-" -f 1`
+          echo "retail_app_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Get commerce-sdk-react version
+        run: |-
+          version=`cat packages/commerce-sdk-react/package.json | jq -r ".version" | cut -d "-" -f 1`
+          echo "commerce_sdk_react_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+
+      - name: Install Monorepo Dependencies
+        run: |-
+          # Install node dependencies
+          node ./scripts/gtime.js monorepo_install npm ci
+
+      - name: Create Release Branch
+        run: git switch -c release-nightly-${{ env.monorepo_version_major }}.x
+
+      - name: Pull latest changes from develop
+        run: git merge develop
+
+      - name: Bump version (monorepo)
+        run: |-
+          npm run bump-version -- "${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Bump version (retail-react-app)
+        run: |-
+          npm run bump-version:retail-react-app -- "${{ env.retail_app_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Bump version (commerce-sdk-react)
+        run: |-
+          npm run bump-version:commerce-sdk-react -- "${{ env.commerce_sdk_react_version_base }}-nightly-${{ env.release_timestamp }}"
+
+      - name: Push branch to origin (Publish to NPM)
+        run: |-
+          git config --global user.name ${{ secrets.GIT_CONFIG_USERNAME }}
+          git config --global user.email ${{ secrets.GIT_CONFIG_EMAIL }}
+          git commit -am "Release ${{ env.monorepo_version_base }}-nightly-${{ env.release_timestamp }}"
+          git push --set-upstream origin release-nightly-${{ env.monorepo_version_major }}.x
```

Comment:
@shethj How does the nightly release get published to npm? Is this the line that triggers it?

In this case, pushing to a release-* branch would eventually reach this part of [workflows/test.yml](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/a7f7d6810e17cc9547ec0848cf4efe81792fd16d/.github/workflows/test.yml#L106-L108)... is that right? I'm trying to understand how the different workflows relate to each other.

If the answer is yes, then I imagine we'll need to update our publish-to-npm.js script too.

Reply:
@vmarta Based on the discussions we had on slack about maintaining a single branch for nightlies.
I updated the script to use a single branch per major version like `release-nightly-3.x` it infers major version from the version mentioned in root `package.json`. It'll automatically create a new branch for v4 whenever that happens.

I also updated our publish-to-npm.js script. This script already has some logic that extracts package name based on   `release-<package-name>-version` so here since the nightly branches are named `release-nightly-3.x`, the package name will == `nightly` and if that is the case, we release with npm tag `nightly-next` instead of `next`